spock-core/src/main/java/org/spockframework/util/ReflectionUtil.java

```diff
@@ -275,4 +277,44 @@ private static void copyField(Field field, Object source, Object target) {
       }
       field.setAccessible(accessible);
   }
+
+  public static <T> T newInstance(Class<T> clazz, Object... args) {
+    Preconditions.notNull(clazz, "Class must not be null");
+    Preconditions.notNull(args, "Argument array must not be null");
+    Preconditions.containsNoNullElements(args, "Individual arguments must not be null");
+
+    try {
+      Class<?>[] parameterTypes = Arrays.stream(args).map(Object::getClass).toArray(Class[]::new);
+      return newInstance(clazz.getDeclaredConstructor(parameterTypes), args);
+    }
+    catch (Throwable t) {
+      throw ExceptionUtils.throwAsUncheckedException(getUnderlyingCause(t));
+    }
+  }
+
+  public static <T> T newInstance(Constructor<T> constructor, Object... args) {
+    Preconditions.notNull(constructor, "Constructor must not be null");
+
+    try {
+      return makeAccessible(constructor).newInstance(args);
+    }
+    catch (Throwable t) {
+      throw ExceptionUtils.throwAsUncheckedException(getUnderlyingCause(t));
+    }
+  }
+
+  @SuppressWarnings("deprecation") // "AccessibleObject.isAccessible()" is deprecated in Java 9
```

Comment:
Right now our target is still 8, we'll have to revisit it when it gets removed and makes problems. As long as it just throws on newer versions, it will implicitly require a public constructor as you said.